e2e/utils/global-setup.ts

```diff
@@ -57,8 +58,23 @@ export default async function (globalConfig: Config.ConfigGlobals) {
     !existsSync('./build')
   ) {
     console.log('Publishing packages to local registry');
-    execSync('pnpm nx-release --local major', {
-      stdio: isVerbose ? 'inherit' : 'ignore',
+    await new Promise<void>((res, rej) => {
+      const publishProcess = exec('pnpm nx-release --local major');
+      let logs = Buffer.from('');
+      if (isVerbose) {
+        publishProcess?.stdout?.pipe(process.stdout);
+        publishProcess?.stderr?.pipe(process.stderr);
+      } else {
+        publishProcess?.stdout?.on('data', (data) => (logs += data));
+        publishProcess?.stderr?.on('data', (data) => (logs += data));
+      }
+      publishProcess.on('exit', (code) => {
+        if (!isVerbose && code && code > 0) {
+          console.log(logs.toString());
+          rej(code);
```

Comment:
This should move outside of the `if` because even if its `verbose`, we should still exit.

Reply:
changed